<?xml version="1.0"?>

<launch>
      <arg name="robot_type" default="aliengo" />
      <arg name="ns" default="" />

      <group ns="$(arg ns)">
            <param name="legged_robot_description"
                  command="$(find xacro)/xacro $(find multi_legged_controllers)/urdf/robot_with_prefix.xacro 
                  ns:=$(arg ns) robot_type:=$(arg robot_type) sim:=true prefix:=$(arg ns)_" />

            <node name="generate_urdf" pkg="multi_legged_controllers" type="generate_urdf.sh"
                  output="screen"
                  args="$(find multi_legged_controllers)/urdf/robot_with_prefix.xacro $(arg robot_type) $(arg ns) $(arg ns)_ true" />

            <rosparam file="$(find legged_gazebo)/config/default.yaml" command="load" />
            <rosparam param="gazebo/imus/base_imu/frame_id" subst_value="True">$(arg ns)_base_imu</rosparam>
            <rosparam param="gazebo/contacts" subst_value="True">[ "$(arg ns)_LF_FOOT",
                  "$(arg ns)_LH_FOOT",
                  "$(arg ns)_RF_FOOT", "$(arg ns)_RH_FOOT" ]</rosparam>
      </group>


</launch>